### ISSUE TITLE
Make v1/sessions#saml_callback rescue more specific

### ISSUE THREAD

**Issue by jholton** [department-of-veterans-affairs/va.gov-team#15236]
I was trying to test [Bennie's PR](https://github.com/department-of-veterans-affairs/vets-api/pull/5151) to see how the end result of an error being raised would manifest itself to the end user.

I changed my local code to ensure that the `ArgumentError` exception would be raised.

I found that it was being swallowed by `saml_callback`.

What happens?
Upon login, `v1/sessions_controller#saml_callback` is invoked.
That method invokes [`set_sso_cookie`](https://github.com/department-of-veterans-affairs/vets-api/blob/36e3489b195b0197fb2e543d7c3998195e4c6c7f/app/controllers/v1/sessions_controller.rb#L316), which invokes [`sso_cookie_content`](https://github.com/department-of-veterans-affairs/vets-api/blob/36e3489b195b0197fb2e543d7c3998195e4c6c7f/app/controllers/concerns/authentication_and_sso_concerns.rb#L121) which triggers a call to MVI when `current_user` attributes are referenced.

In this manufactured example, an [`ArgumentError`](https://github.com/department-of-veterans-affairs/vets-api/pull/5151/files#diff-25a31945f46039020dc93bbceff47aabd974d1bda0f697cdd72b82468514192bR42) is raised and is subsequently [`rescue`d back in #saml_callback](https://github.com/department-of-veterans-affairs/vets-api/blob/36e3489b195b0197fb2e543d7c3998195e4c6c7f/app/controllers/v1/sessions_controller.rb#L61).
Then as part of handling the error, the [error's `code` is referenced](https://github.com/department-of-veterans-affairs/vets-api/blob/36e3489b195b0197fb2e543d7c3998195e4c6c7f/app/controllers/v1/sessions_controller.rb#L267)

Well, `ArgumentError` does not have a 'code' attribute and another exception is raised: 
`NoMethodError: undefined method 'code' for #<ArgumentError: required value missing>`

I searched Sentry and don't see any occurrences of "undefined method 'code'" or a `DoubleRenderError` for the `v1/sessions` controller or [any other "unexpected" errors](http://sentry.vfs.va.gov/organizations/vsp/issues/?environment=production&project=3&query=transaction%3AV1%3A%3ASessionsController%23saml_callback&statsPeriod=90d), but I think my point is still valid:

That `rescue` could be swallowing/hiding any error that occur during the `#saml_callback` call.

Feel free to determine that this is not an issue. 😄 
Just wanted to get this written up for consideration.

**Comment by joeniquette** [comment]
@ericbuckley I believe this statement is related to the issue you brought up to me last week when we were discussing Sentry errors. If we are resolving this with another ticket do you mind linking it here?

**Comment by ericbuckley** [comment]
Tldr; lets just delete the call to `set_sso_cookie`

@joeniquette I haven't started any work on this issue, nor do I think @bramleyjl has.  However I think this issue is a bit misleading and should be replaced with another.

a) We should NEVER be setting a SSO cookie when authenticating with SSOe, as we will be sending users back to MHV via the eauth proxy and the old SSO cookie is no longer needed
b) even if we did need to set the cookie, it doesn't make sense that we'd want to invoke MPI for any data.  We just received a response from SSOe with ALL the MPI data in it.  In other words just look in the response XML for the data, don't query MPI again.
c) the way the code is currently written, its possible that this MPI call is made BEFORE we uplevel, meaning we don't yet know that the user even has an MPI record.  I believe, even if `a` and `b` above were wrong, we'd want the set_sso_cookie call to happen only after upleveling

**Comment by bramleyjl** [comment]
Duplicate issue: https://github.com/department-of-veterans-affairs/va.gov-team/issues/15883
PR (linked to duplicate issue): https://github.com/department-of-veterans-affairs/vets-api/pull/5437
